nss-ice-phantoms/BangazonSite
# Meg current order

# Description
- Added Past Orders view `/Orders` #17
- Added Cart View `/Orders/Cart` (or click on cart icon)
- Added ability to complete order by selecting payment method #8 
- Added ability to delete order (aka "empty cart") #10
- Added logic to ensure all URLs work accordingly (users can only see their specific orders, users cannot edit or delete a past order, etc.)

Still need to do:
- Add order total to past orders and current order
- Add "View Order History" link to account settings
- Add confirmation/thank you screen when order is placed

## Type of change

# Testing Instructions

1. `git fetch --all`
1. `git checkout meg-current-order`
1. Run Bangazon
1. DO NOT LOG IN. Go to these pages and make sure you can't access them:
   `/Orders`
   `/Orders/Cart`
   `/Orders/Details/1`
   `/Orders/Edit/1`
   `/Orders/Delete/1`
1. Now log in as admin: `admin@admin.com` `Admin8*`
1. Add some extra Orders and OrderProducts to the database (see below). These are tied to the admin user. (Make sure the `OrderId` of each OrderProduct matches the orders created in the database.)
1. Go to `/Orders` and you should (most likely) see 2 orders, both with payment types.
1. Click on "Details" for an order and you'll see that order's products. (No delete or edit.)
1. Click on the cart icon and you should only see 1 order, i.e. your current shopping cart, along with its products.
1. Click "Proceed to Checkout," select a payment type, click "Submit Order," and you should then be redirected to the "Order History" page, where you'll see your just-submitted order, with its completion date and payment method listed.
1. If you now click on the cart icon, it should tell you the cart is empty.
1. Go back to the SQL below and add the second order again (without the PaymentTypeId). Note the order's ID, then use it to add more OrderProducts.
1. Go back to your cart and it should have items in it again. Click "Empty Shopping Cart."
1. Click "Yes" and the cart (i.e. order) along with its corresponding OrderProducts will be deleted.
1. If you want to test out user authentication, create a new user, then run the order SQL below but replace the `UserId` with the new user's Id. When you view your cart/past orders, you should only see your orders, and not the admin's.

```
INSERT into [Order] (DateCreated, DateCompleted, UserId, PaymentTypeId)
VALUES ('10 April 2019', '11 April 2019', '536e260c-1b83-445c-9c05-e89372cb77ad', 1)

INSERT into [Order] (DateCreated, UserId)
VALUES ('01 April 2019', '536e260c-1b83-445c-9c05-e89372cb77ad')
```
```
INSERT into OrderProduct (OrderId, ProductId) VALUES (2, 1)
INSERT into OrderProduct (OrderId, ProductId) VALUES (2, 2)
INSERT into OrderProduct (OrderId, ProductId) VALUES (3, 1)
INSERT into OrderProduct (OrderId, ProductId) VALUES (3, 2)
```


# Checklist:
